compiler/modulegraphs.nim

```diff
@@ -75,6 +75,8 @@ type
     typeInstCache*: Table[ItemId, seq[LazyType]] # A symbol's ItemId.
     procInstCache*: Table[ItemId, seq[LazyInstantiation]] # A symbol's ItemId.
     attachedOps*: array[TTypeAttachedOp, Table[ItemId, LazySym]] # Type ID, destructors, etc.
+    typeBoundOps*: Table[ItemId, TStrTable]
```

Comment:
Can it be done without this lookup table?

Reply:
It should be possible, and we should have the option of storing it in the type symbol rather than the type, because of the way we reduce to nominal types. I put it here initially because of the relation to scopes. Maybe there is an alternative to using `TStrTable` as well but it seems fitting for me.

Not sure which makes more sense between putting it in `TType` or `TSym`, `TStrTable` is 24 bytes without pointer indirection and not every type has it, and in `TSym` we can use an [object variant branch](https://github.com/nim-lang/Nim/blob/e69eb99a159859338dbcd9606cf31a6c4700d90b/compiler/ast.nim#L701) for `skType` to reuse some bytes. But whichever one makes more sense for the compiler's design is more important.

Reply:
This point is this table is nowwhere required, you simply query the interface table of the module that the type was declared in. This is not about efficiency or its lack thereof, the way I describe it would work automatically with IC.